addons/mrp/models/mrp_production.py

```diff
@@ -1854,7 +1856,24 @@ def _default_amounts(production):
                     new_moves_vals.append(move_vals)
                     moves.append(move)
 
-        backorder_moves = self.env['stock.move'].create(new_moves_vals)
+        # Merge move vals for the same product from the same source if no explicit amounts were provided
+        updated_moves_vals = {}
+        if not amounts_arg:
+            for move_vals in new_moves_vals:
+                key_tuple = (move_vals['product_id'], move_vals['location_id'])
+                if key_tuple in updated_moves_vals:
+                    if 'quantity' in updated_moves_vals[key_tuple]:
+                        updated_moves_vals[key_tuple]['quantity'] += move_vals.get('quantity', 0)
+                    else:
+                        updated_moves_vals[key_tuple]['quantity'] = move_vals.get('quantity', 0)
+                    if 'product_uom_qty' in updated_moves_vals[key_tuple]:
+                        updated_moves_vals[key_tuple]['product_uom_qty'] += move_vals.get('product_uom_qty', 0)
+                    else:
+                        updated_moves_vals[key_tuple]['product_uom_qty'] = move_vals.get('product_uom_qty', 0)
+                else:
+                    updated_moves_vals[key_tuple] = move_vals
+
+        backorder_moves = self.env['stock.move'].create(list(updated_moves_vals.values()) or new_moves_vals)
```

Comment:
since we remove moves, it seems that it would break the logic below, that zip moves to backorder_moves (so after the first removed element, we would have all other elements associated to the wrong move), but I'm not sure what this break and how to fix that